etl-pipeline/mappings/publisher_backlist.py

```diff
@@ -44,6 +60,90 @@ def applyFormatting(self):
 
         self.record.rights = self.format_rights()
 
+    def get_hathi_id(self) -> Optional[str]:
+        hath_identifier = next((identifier for identifier in self.record.identifiers if identifier.endswith('hathi')), None)
+
+        if hath_identifier is not None:
+            return hath_identifier.split('|')[0]
+
+        return None
+
+    def add_has_part(self) -> str:
+        record_permissions = self.parse_permissions(self.source.get('Access type in DRB (from Access types)')[0])
+        file_location = self.source.get('DRB_File Location')
+        destination_file_bucket = os.environ['FILE_BUCKET'] if not record_permissions['requires_login'] else f"drb-files-limited-{os.environ['ENVIRONMENT']}"
+        pdf_bucket = os.environ["PDF_BUCKET"]
+        hathi_id = self.get_hathi_id()
+
+        if not file_location and not hathi_id:
+            raise Exception(f'Unable to get file url for {self.record}')
+        
+        self.record.has_part.append(str(Part(
+            index=1,
+            url=get_stored_file_url(
+                storage_name=os.environ['FILE_BUCKET'],
+                file_path=f"manifests/publisher_backlist/{self.record.source}/{self.record.source_id}.json",
+            ),
+            source=self.record.source,
+            file_type='application/webpub+json',
+            flags=str(
+                FileFlags(
+                    reader=True, 
+                    nypl_login=record_permissions['requires_login'], 
+                    fulfill_limited_access=record_permissions['requires_login']
+                )
+            ),
+        )))
+
+        file_path = f"titles/publisher_backlist/{self.record.source}/{self.record.source_id}.pdf"
+        
+        if hathi_id:
+            self.record.has_part.append(str(Part(
+                index=1,
+                url=get_stored_file_url(
+                    storage_name=destination_file_bucket,
+                    file_path=file_path,
+                ),
+                source=self.record.source,
+                file_type='application/pdf',
+                flags=str(
+                    FileFlags(
+                        download=record_permissions['is_downloadable'], 
+                        nypl_login=record_permissions['requires_login'], 
+                        fulfill_limited_access=record_permissions['requires_login']
+                    )
+                ),
+                source_url=f'https://{pdf_bucket}.s3.amazonaws.com/tagged_pdfs/{hathi_id}.pdf'
```

Comment:
`get_stored_file_url`?

Reply:
urgh I think it's always going to be in the pdf_bucket !

Reply:
I see, because when running locally, we'll still want to grab from the remote bucket?

Reply:
I think so! It won't really work tho :(